src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +45,43 @@
      * Андрей - 600
      * София - 900</p>
      */
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+        LocalDate startDate = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate endDate = LocalDate.parse(dateTo, FORMATTER);
+        if (!endDate.isAfter(startDate)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        StringBuilder salaryInfo = new StringBuilder("Отчёт за период "
+                + dateFrom + " - " + dateTo + "\n");
+        List<Employee> employees = new ArrayList<>();
+
+        for (String name : names) {
+            Employee employee = new Employee(name);
+            employees.add(employee);
+            for (String i : data) {
+                String[] dataInformation = i.split(" ");
```

Comment:
_dataInformation_ is not a very abstract variable name. Can we rename this variable?

Reply:
done